sycl/include/CL/sycl/accessor.hpp

```diff
@@ -382,24 +392,29 @@ class image_accessor
   };
 
   //  Available only when: (accessTarget == access::target::image ||
-  //  accessTarget == access::target::host_image) && accessMode ==
-  //  access::mode::read
+  //  accessTarget == access::target::host_image) && (accessMode ==
+  //  access::mode::read || accessMode == access::mode::read_write)
+  //  The appropriate allowed AccessMode for the AccessTarget is checked in the
+  //  static_assert in the class.
   template <typename CoordT, int Dims = Dimensions,
             typename = detail::enable_if_t<
                 (Dims > 0) && (IsValidCoordDataT<Dims, CoordT>::value) &&
-                (IsImageAcc || IsHostImageAcc) && IsImageAccessRead>>
+                (IsImageAcc || IsHostImageAcc) && IsImageAccessAnyRead>>
   DataT read(const CoordT &Coords) const {
     // TODO: To be implemented.
     throw cl::sycl::feature_not_supported("Read API is not implemented.");
     return;
   };
 
-  // Available only when: (accessTarget == access::target::image || accessTarget
-  // == access::target::host_image) && accessMode == access::mode::read
+  //  Available only when: (accessTarget == access::target::image ||
+  //  accessTarget == access::target::host_image) && (accessMode ==
+  //  access::mode::read || accessMode == access::mode::read_write)
+  //  The appropriate allowed AccessMode for the AccessTarget is checked in the
+  //  static_assert in the class.
   template <typename CoordT, int Dims = Dimensions,
             typename = detail::enable_if_t<
                 (Dims > 0) && (IsValidCoordDataT<Dims, CoordT>::value) &&
-                (IsImageAcc || IsHostImageAcc) && IsImageAccessRead>>
+                (IsImageAcc || IsHostImageAcc) && IsImageAccessAnyRead>>
```

Comment:
`read_write` access mode must be available with the host accessor.

Reply:
added the requested changes.